modules/ROOT/pages/type-definitions/directives/custom-logic.adoc

```diff
@@ -0,0 +1,641 @@
+
+=== Definition
+
+[source, graphql, indent=0]
+----
+"""Int | Float | String | Boolean | ID | DateTime | Enum"""
+scalar ScalarOrEnum
+
+"""Instructs @neo4j/graphql to wrap the property in a coalesce() function during queries, using the single value specified."""
+directive @coalesce(
+    """The value to use in the coalesce() function. Must be a scalar type and must match the type of the field with which this directive decorates."""
+    value: Scalar!,
+) on FIELD_DEFINITION
+----
+
+=== Usage
+
+`@coalesce` may be used with enums. 
+When setting the default value for an enum field, it must be one of the enumerated enum values:
+
+[source, graphql, indent=0]
+----
+enum Status {
+    ACTIVE
+    INACTIVE
+}
+type Movie {
+    status: Status @coalesce(value: ACTIVE)
+}
+----
+
+[[type-definitions-default-values-limit]]
+== `@limit`
+
+Available on nodes, this directive injects values into a query such as the `limit`.
+
+=== Definition
+
+[source, graphql, indent=0]
+----
+"""The `@limit` is to be used on nodes, where applied will inject values into a query such as the `limit`."""
+directive @limit(
+    default: Int
+    max: Int
+) on OBJECT
+----
+
+=== Usage
+
+The directive has two arguments:
+
+* `default` - if no `limit` argument is passed to the query, the default limit is used. 
+The query may still pass a higher or lower `limit`.
+* `max` - defines the maximum limit to be passed to the query. 
+If a higher value is passed, it is used instead. 
+If no `default` value is set, `max` is used for queries without limit.
+
+[source, graphql, indent=0]
+----
+{
+  Movie @limit(amount: 5) {
+    title
+    year
+  }
+}
+----
+
+== `@customResolver`
+
+The Neo4j GraphQL Library generates query and mutation resolvers, so you don't need to implement them yourself.
+However, if you need additional behaviors besides the autogenerated CRUD operations, you can specify custom resolvers for these scenarios.
+This page describes how to set them up.
+
+To add a field to an object type which is resolved from existing values in the type, rather than storing new values, you should mark it with the `@customResolver` directive and define a custom resolver for it.
+
+Take, for instance, this schema:
+
+[source, javascript, indent=0]
+----
+const typeDefs = `
+    type User {
+        firstName: String!
+        lastName: String!
+        fullName: String! @customResolver(requires: "firstName lastName")
+    }
+`;
+
+const resolvers = {
+    User: {
+        fullName(source) {
+            return `${source.firstName} ${source.lastName}`;
+        },
+    },
+};
+
+const neoSchema = new Neo4jGraphQL({
+    typeDefs,
+    resolvers,
+});
+----
+
+Here `fullName` is a value that is resolved from the fields `firstName` and `lastName`. 
+Specifying the `@customResolver` directive on the field definition keeps `fullName` from being included in any `Query` or `Mutation` fields and hence as a property on the `:User` node in the database.
```

Comment:
Bringing back the discussion about capitalizing terms like query and mutation or using this formatting.

Reply:
i used lower case, following graphql.org and apollographql.com